src/common/src/types/from_sql.rs

```diff
@@ -12,55 +12,171 @@
             _ => bail_not_implemented!("the postgres decoding for {ty} is unsupported"),
         })
     }
 
-    fn accepts(ty: &Type) -> bool {
+    fn accepts(ty: &postgres_types::Type) -> bool {
         matches!(
             *ty,
-            Type::BOOL
-                | Type::INT2
-                | Type::INT4
-                | Type::INT8
-                | Type::FLOAT4
-                | Type::FLOAT8
-                | Type::DATE
-                | Type::TIME
-                | Type::TIMESTAMP
-                | Type::TIMESTAMPTZ
-                | Type::JSONB
-                | Type::INTERVAL
-                | Type::BYTEA
-                | Type::VARCHAR
-                | Type::TEXT
+            postgres_types::Type::BOOL
+                | postgres_types::Type::INT2
+                | postgres_types::Type::INT4
+                | postgres_types::Type::INT8
+                | postgres_types::Type::FLOAT4
+                | postgres_types::Type::FLOAT8
+                | postgres_types::Type::DATE
+                | postgres_types::Type::TIME
+                | postgres_types::Type::TIMESTAMP
+                | postgres_types::Type::TIMESTAMPTZ
+                | postgres_types::Type::JSONB
+                | postgres_types::Type::INTERVAL
+                | postgres_types::Type::BYTEA
+                | postgres_types::Type::VARCHAR
+                | postgres_types::Type::TEXT
         )
     }
 }
+
+/// The following table shows the mapping between Rust types and Sql Server types in tiberius.
+/// |Rust Type|Sql Server Type|
+/// |`u8`|`tinyint`|
+/// |`i16`|`smallint`|
+/// |`i32`|`int`|
+/// |`i64`|`bigint`|
+/// |`f32`|`float(24)`|
+/// |`f64`|`float(53)`|
+/// |`bool`|`bit`|
+/// |`String`/`&str`|`nvarchar`/`varchar`/`nchar`/`char`/`ntext`/`text`|
+/// |`Vec<u8>`/`&[u8]`|`binary`/`varbinary`/`image`|
+/// |[`Uuid`]|`uniqueidentifier`|
+/// |[`Numeric`]|`numeric`/`decimal`|
+/// |[`Decimal`] (with feature flag `rust_decimal`)|`numeric`/`decimal`|
+/// |[`XmlData`]|`xml`|
+/// |[`NaiveDateTime`] (with feature flag `chrono`)|`datetime`/`datetime2`/`smalldatetime`|
+/// |[`NaiveDate`] (with feature flag `chrono`)|`date`|
+/// |[`NaiveTime`] (with feature flag `chrono`)|`time`|
+/// |[`DateTime`] (with feature flag `chrono`)|`datetimeoffset`|
+///
+/// See the [`time`] module for more information about the date and time structs.
+///
+/// [`Row#get`]: struct.Row.html#method.get
+/// [`Row#try_get`]: struct.Row.html#method.try_get
+/// [`time`]: time/index.html
+/// [`Uuid`]: struct.Uuid.html
+/// [`Numeric`]: numeric/struct.Numeric.html
+/// [`Decimal`]: numeric/struct.Decimal.html
+/// [`XmlData`]: xml/struct.XmlData.html
+/// [`NaiveDateTime`]: time/chrono/struct.NaiveDateTime.html
+/// [`NaiveDate`]: time/chrono/struct.NaiveDate.html
+/// [`NaiveTime`]: time/chrono/struct.NaiveTime.html
+/// [`DateTime`]: time/chrono/struct.DateTime.html
+impl<'a> tiberius::FromSql<'a> for ScalarImpl {
```

Comment:
We cannot do that, since, in `connector` crate, both `tiberius::FromSql` and `ScalarImpl` are in third-party modules.

Reply:
You may use the newtype pattern to impl 3rd party trait for 3rd party type:
```rust
type FromTiberius(ScalarImpl);

impl<'a> tiberius::FromSql<'a> for FromTiberius {...}
```

+1 to avoid putting too specialized stuff in `common`.

Reply:
Marking as unresolved and second (or "third") this. These SQL Server specific stuff shall not be in `common` and the correct way to follow the orphan rule is to use newtype pattern.

Reply:
updated